In [11]:
import pandas as pd
import numpy as np


In [12]:
df = pd.read_csv('member_export.csv', encoding="ISO-8859-1", low_memory=False)

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22645 entries, 0 to 22644
Data columns (total 85 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   expid       22645 non-null  object 
 1   membid      22645 non-null  int64  
 2   peakid      22645 non-null  object 
 3   myear       22645 non-null  int64  
 4   mseason     22645 non-null  int64  
 5   fname       22617 non-null  object 
 6   lname       21949 non-null  object 
 7   sex         22645 non-null  object 
 8   age         22645 non-null  int64  
 9   birthdate   22645 non-null  object 
 10  yob         21308 non-null  float64
 11  calcage     22645 non-null  int64  
 12  citizen     22645 non-null  object 
 13  status      22642 non-null  object 
 14  residence   21322 non-null  object 
 15  occupation  13820 non-null  object 
 16  leader      22645 non-null  bool   
 17  deputy      22645 non-null  bool   
 18  bconly      22645 non-null  bool   
 19  nottobc     22645 non-nul

In [14]:
# The dataset has Mallory and Irvine, so remove them
# this command removes "hired" as Sherpa deaths happen for different reasons
# Also remove misc items that are not relevant
north_climbs = df[(df['hired']==False) & (df['route1'].str[0]=='N') & (df['mperhighpt']>=8000) &
                 (df['fname']!='Andrew Comyn (Sandy)') & (df['fname']!='George Herbert Leigh') & 
                 (df['mski'] ==False) & (df['route1']!='N Col (training, recon)')]

In [15]:
# a dataset for the Sherpa

north_climbs_h = df[ (df['route1'].str[0]=='N') & (df['mperhighpt']>=8000) &
                 (df['fname']!='Andrew Comyn (Sandy)') & (df['fname']!='George Herbert Leigh') & 
                 (df['mski'] ==False) & (df['route1']!='N Col (training, recon)')]

In [16]:
# just use the columns we are interested in.  "msuccess" is "member success" and means the climber reached the summit
north_climbs = north_climbs[['expid', 'fname', 'lname', 'sex', 'yob', 'calcage', 'hired', 'sherpa', 'msuccess', 'mo2used', 'mperhighpt', 'death', 'deathhgtm',  'route1']]
# just use the columns we are interested in.  "msuccess" is "member success" and means the climber reached the summit
north_climbs_h = north_climbs_h[['expid', 'fname', 'lname', 'sex', 'yob', 'calcage', 'hired', 'sherpa', 'msuccess', 'mo2used', 'mperhighpt', 'death', 'deathhgtm',  'route1']]


In [17]:
north_climbs.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2835 entries, 42 to 21913
Data columns (total 14 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   expid       2835 non-null   object 
 1   fname       2831 non-null   object 
 2   lname       2636 non-null   object 
 3   sex         2835 non-null   object 
 4   yob         2657 non-null   float64
 5   calcage     2835 non-null   int64  
 6   hired       2835 non-null   bool   
 7   sherpa      2835 non-null   bool   
 8   msuccess    2835 non-null   bool   
 9   mo2used     2835 non-null   bool   
 10  mperhighpt  2835 non-null   int64  
 11  death       2835 non-null   bool   
 12  deathhgtm   2835 non-null   int64  
 13  route1      2835 non-null   object 
dtypes: bool(5), float64(1), int64(3), object(5)
memory usage: 235.3+ KB


In [18]:
pd.set_option('max_columns', 10)

In [19]:
# break down of summits vs non-summits for all climbs
north_climbs.groupby(by=['msuccess']).count()

expid  fname  lname   sex   yob  ...  mo2used  mperhighpt  death  \
msuccess                                   ...                               
False       953    949    878   953   902  ...      953         953    953   
True       1882   1882   1758  1882  1755  ...     1882        1882   1882   

          deathhgtm  route1  
msuccess                     
False           953     953  
True           1882    1882  

[2 rows x 13 columns]

In [20]:
# do some sanity checks to make sure the probability of a summit increase the higher you make it

In [21]:
# First Step
north_climbs[(north_climbs['mperhighpt'] >=8500)].groupby(by=['msuccess']).count()


expid  fname  lname   sex   yob  ...  mo2used  mperhighpt  death  \
msuccess                                   ...                               
False       448    444    429   448   418  ...      448         448    448   
True       1882   1882   1758  1882  1755  ...     1882        1882   1882   

          deathhgtm  route1  
msuccess                     
False           448     448  
True           1882    1882  

[2 rows x 13 columns]

In [22]:
1882/(448+1882)

0.8077253218884121

In [23]:
# Second Step
north_climbs[(north_climbs['mperhighpt'] >=8600)].groupby(by=['msuccess']).count()


expid  fname  lname   sex   yob  ...  mo2used  mperhighpt  death  \
msuccess                                   ...                               
False       252    251    242   252   243  ...      252         252    252   
True       1882   1882   1758  1882  1755  ...     1882        1882   1882   

          deathhgtm  route1  
msuccess                     
False           252     252  
True           1882    1882  

[2 rows x 13 columns]

In [24]:
1882/(1882+252)

0.8819119025304593

In [25]:
# Third Step
north_climbs[(north_climbs['mperhighpt'] >=8650)].groupby(by=['msuccess']).count()


expid  fname  lname   sex   yob  ...  mo2used  mperhighpt  death  \
msuccess                                   ...                               
False       150    150    145   150   143  ...      150         150    150   
True       1882   1882   1758  1882  1755  ...     1882        1882   1882   

          deathhgtm  route1  
msuccess                     
False           150     150  
True           1882    1882  

[2 rows x 13 columns]

In [26]:
1882/(1882+150)

0.9261811023622047

In [27]:
# Look at what happens when you turn around between First and Second Step

# First Step
north_climbs[(north_climbs['mperhighpt'] >=8500) & (north_climbs['mperhighpt'] <=8600)].groupby(by=['death']).count()

expid  fname  lname  sex  yob  ...  msuccess  mo2used  mperhighpt  \
death                                 ...                                  
False    284    280    270  284  261  ...       284      284         284   
True      10     10     10   10   10  ...        10       10          10   

       deathhgtm  route1  
death                     
False        284     284  
True          10      10  

[2 rows x 13 columns]

 Assume nothing other than that they reached the "death zone" and died
 $$P(Summit|Reached\, 8000m)$$

In [28]:
death_zone = north_climbs[ (north_climbs['death']==True)]
death_zone.groupby(by=['msuccess']).count()

expid  fname  lname  sex  yob  ...  mo2used  mperhighpt  death  \
msuccess                                 ...                               
False        24     24     24   24   22  ...       24          24     24   
True         33     33     31   33   32  ...       33          33     33   

          deathhgtm  route1  
msuccess                     
False            24      24  
True             33      33  

[2 rows x 13 columns]

In [29]:
(33)/(33+24)

0.5789473684210527

$P(Summit|Reached\, 8000m) = \frac{33}{33+24} = 58\%$

Using only that we know they reached the location of the ice axe (8840m) -- it does not assume it was either the ascent or descent



In [30]:
north_climbs[(north_climbs['death'] == True) & (north_climbs['mperhighpt'] >=8440)].groupby(by=['msuccess']).count()

expid  fname  lname  sex  yob  ...  mo2used  mperhighpt  death  \
msuccess                                 ...                               
False        16     16     16   16   16  ...       16          16     16   
True         33     33     31   33   32  ...       33          33     33   

          deathhgtm  route1  
msuccess                     
False            16      16  
True             33      33  

[2 rows x 13 columns]

In [31]:
(33/(16+33))

0.673469387755102

$P(Summit|Reached\, 8440m) = \frac{33}{33+16} = 67\%$

And the theory that they died at the ice axe site on the ascent which I will calculate at the percentage of people who die on the ascent between 8000 and 8500 meters

In [32]:
ascent_below_first_step = north_climbs[(north_climbs['death'] == True) & (north_climbs['mperhighpt'] <= 8500 ) & (north_climbs['mperhighpt'] == north_climbs['deathhgtm'])]
ascent_below_first_step.groupby(by=['mperhighpt']).count()  # we really just need the total count, but this will break out where they died

expid  fname  lname  sex  yob  ...  msuccess  mo2used  death  \
mperhighpt                                 ...                             
8100            1      1      1    1    0  ...         1        1      1   
8230            2      2      2    2    2  ...         2        2      2   
8250            1      1      1    1    1  ...         1        1      1   
8300            2      2      2    2    1  ...         2        2      2   
8500            3      3      3    3    3  ...         3        3      3   

            deathhgtm  route1  
mperhighpt                     
8100                1       1  
8230                2       2  
8250                1       1  
8300                2       2  
8500                3       3  

[5 rows x 13 columns]

A total of 9 people out of 2835 climbs died on the ascent below the First Step

Next we look at death on descent after having reached at least the ice axe location

In [33]:
death_on_descent = north_climbs[(north_climbs['death'] == True)  & (north_climbs['mperhighpt'] >=8440) & (north_climbs['mperhighpt'] > north_climbs['deathhgtm'])]
death_on_descent.groupby(by=['msuccess']).count()

expid  fname  lname  sex  yob  ...  mo2used  mperhighpt  death  \
msuccess                                 ...                               
False        11     11     11   11   11  ...       11          11     11   
True         33     33     31   33   32  ...       33          33     33   

          deathhgtm  route1  
msuccess                     
False            11      11  
True             33      33  

[2 rows x 13 columns]

In [34]:
33/(33+11)

0.75

$P(Summit|Died\,on\,Descent) = \frac{35}{35+14} = 75\%$

In [35]:
# Lets look at people who turn around between the Second and Third Steps
turn_around_between_second_third = north_climbs[(north_climbs['mperhighpt']>=8600) &(north_climbs['mperhighpt']<=8650)]
turn_around_between_second_third.groupby(by=['death']).count()

expid  fname  lname  sex  yob  ...  msuccess  mo2used  mperhighpt  \
death                                 ...                                  
False    130    129    125  130  127  ...       130      130         130   
True       4      4      4    4    4  ...         4        4           4   

       deathhgtm  route1  
death                     
False        130     130  
True           4       4  

[2 rows x 13 columns]

In [36]:
# for the 149 climbers who turned around after reaching the Third Step, lets look at the results
# Note 1 person died above the third step on the ascent -- that is why it does not match the 150 above


third_step_turn_around = north_climbs[(north_climbs['mperhighpt']>=8650) & (north_climbs['msuccess']==False) & (north_climbs['mperhighpt'] > north_climbs['deathhgtm'])]
third_step_turn_around.groupby(by=['death']).count()

expid  fname  lname  sex  yob  ...  msuccess  mo2used  mperhighpt  \
death                                 ...                                  
False    144    144    139  144  137  ...       144      144         144   
True       5      5      5    5    5  ...         5        5           5   

       deathhgtm  route1  
death                     
False        144     144  
True           5       5  

[2 rows x 13 columns]

In [37]:
# died on ascent above Third Step
north_climbs[(north_climbs['mperhighpt']>=8650) & (north_climbs['msuccess']==False) & (north_climbs['mperhighpt'] == north_climbs['deathhgtm'])]

expid              fname      lname sex     yob  ...  mo2used  \
8680  EVER04109  Mariana Prodanova  Maslarova   F  1961.0  ...     True   

      mperhighpt  death  deathhgtm          route1  
8680        8700   True       8700  N Col-NE Ridge  

[1 rows x 14 columns]

Now compute
$$P(Summit|Reached\,Third\,Step\,and\,died\,on\,descent) $$

In [38]:
at_least_third_step = north_climbs[(north_climbs['mperhighpt']>=8650) & (north_climbs['death']==True) & (north_climbs['mperhighpt'] > north_climbs['deathhgtm'])]
at_least_third_step.groupby(by=['msuccess']).count()

expid  fname  lname  sex  yob  ...  mo2used  mperhighpt  death  \
msuccess                                 ...                               
False         5      5      5    5    5  ...        5           5      5   
True         33     33     31   33   32  ...       33          33     33   

          deathhgtm  route1  
msuccess                     
False             5       5  
True             33      33  

[2 rows x 13 columns]

In [39]:
33/(33+5)

0.868421052631579


$P(Summit|Reached\,Third\,Step\,and\,died\,on\,descent)  = \frac{33}{33+5} = 87\%$  

In [40]:
# verify that including 'hired' does not meaningful chance the results.
# I leave out the hired because their deaths are largely due to the clients mistakes, 
# they carry much larger loads, and use much less oxygen.
# All of their names are listed at the end of this analysis
at_least_third_step_h = north_climbs_h[(north_climbs_h['mperhighpt']>=8650) & (north_climbs_h['death']==True) & (north_climbs_h['mperhighpt'] > north_climbs_h['deathhgtm'])]
at_least_third_step_h.groupby(by=['msuccess']).count()

expid  fname  lname  sex  yob  ...  mo2used  mperhighpt  death  \
msuccess                                 ...                               
False         6      6      6    6    5  ...        6           6      6   
True         35     35     33   35   34  ...       35          35     35   

          deathhgtm  route1  
msuccess                     
False             6       6  
True             35      35  

[2 rows x 13 columns]

In [41]:
# Review the people who died on the descent to check accuracy
# note, this database does not credit David Sharp has having reached the summit
north_climbs[(north_climbs['mperhighpt']>=8650) & (north_climbs['death']==True)]

expid                      fname                lname sex     yob  \
1924   EVER91116                    Junichi             Futagami   M  1951.0   
3239   EVER94103            Michael Anthony          Rheinberger   M  1940.0   
3263   EVER94106         Fang-Fang (Norman)                 Shih   M  1967.0   
4089   EVER96110                    Tsewang               Smanla   M  1957.0   
4107   EVER96110             Tsewang Paljor                  NaN   M  1968.0   
4109   EVER96110                Dorje Morup                  NaN   M  1948.0   
4733   EVER97108                       Ivan            Plotnikov   M  1960.0   
4749   EVER97108                    Nikolai          Chevtchenko   M  1944.0   
5564   EVER98107        Sergei Anatolievich            Arsentiev   M  1958.0   
5565   EVER98107                    Francys  Distefano-Arsentiev   F  1958.0   
5700   EVER98125                Mark Howard             Jennings   M  1948.0   
6403   EVER99104                     Pascal            Debrouwer   M  1970.0   
6598   EVER99131            Tadeusz (Tadek)             Kudelski   M  1954.0   
6619   EVER99134             Vasili (Vazyl)              Kopytko   M  1964.0   
7022   EVER00143                   Geng-Hua                  Yan   M  1958.0   
7383   EVER01126                       Mark              Auricht   M  1963.0   
7572   EVER01148             Aleksei (Alex)            Nikiforov   M  1955.0   
8670   EVER04107             Hristo Ganchev              Hristov   M  1977.0   
8680   EVER04109          Mariana Prodanova            Maslarova   F  1961.0   
8826   EVER04131                      Shoko                  Ota   F  1941.0   
8870   EVER04142                    Mu-Taek                 Park   M  1969.0   
8874   EVER04142                        Min                 Jang   M  1978.0   
9290   EVER05115                      Marko           Lihteneker   M  1969.0   
9457   EVER05142  Sirigereshiva Shankarappa            Chaitanya   M  1973.0   
10091  EVER06134             Jacques-Hughes             Letrange   M  1967.0   
10182  EVER06146                      Vitor              Negrete   M  1967.0   
10458  EVER06186                        Sri               Kishan   M  1979.0   
10478  EVER06189                     Thomas                Weber   M  1964.0   
10504  EVER06189                       Igor            Plyushkin   M  1952.0   
11030  EVER07148                   Shinichi                Ishii   M  1944.0   
11308  EVER07146                 Pierangelo             Maurizio   M  1955.0   
12822  EVER09164                      Frank             Ziebarth   M  1980.0   
12871  EVER09102                   Wen-Hong                   Wu   M     NaN   
13598  EVER11136                       John              Delaney   M  1969.0   
13755  EVER10129                    Hiroshi            Ogasawara   M  1947.0   
14104  EVER10171                      Peter              Kinloch   M  1981.0   
15342  EVER12178                Ralf Dieter               Arnold   M  1970.0   
15344  EVER12179                  Juan Jose         Polo Carbayo   M  1969.0   
21545  EVER19160                      Ernst             Landgraf   M  1954.0   

       ...  mo2used  mperhighpt  death  deathhgtm  \
1924   ...     True        8849   True       8830   
3239   ...     True        8849   True       8680   
3263   ...     True        8849   True       8700   
4089   ...     True        8849   True       8700   
4107   ...     True        8849   True       8550   
4109   ...     True        8849   True       8680   
4733   ...     True        8849   True       8700   
4749   ...     True        8849   True       8700   
5564   ...    False        8849   True       8500   
5565   ...    False        8849   True       8600   
5700   ...     True        8849   True       8200   
6403   ...     True        8849   True       8400   
6598   ...     True        8849   True       8600   
6619   ...    False        8849   True       8500   
7022   ...     

In [42]:
# High point probably not accurate, so, the probability would increase slightly if he was credited with a summit.
north_climbs[(north_climbs['lname']=='Sharp') & (north_climbs['death']==True)]

expid     fname  lname sex     yob  ...  mo2used  mperhighpt  \
10087  EVER06135  David W.  Sharp   M  1972.0  ...     True        8500   

       death  deathhgtm          route1  
10087   True       8500  N Col-NE Ridge  

[1 rows x 14 columns]

In [43]:
# And if Sharp made the summit
34/(34+4)

0.8947368421052632

In [44]:
# And the hero's roster
north_climbs_h[(north_climbs_h['mperhighpt']>=8000) & (north_climbs_h['death']==True) & (north_climbs_h['hired']==True)]

expid             fname   lname sex     yob  ...  mo2used  \
3577   EVER95305            Jangbu  Sherpa   M  1967.0  ...     True   
4838   EVER97125  Mingmar (Mingma)  Tamang   M     NaN  ...     True   
15835  EVER13156           Namgyal  Sherpa   M  1977.0  ...     True   
20451  EVER18166       Pasang Nuru  Sherpa   M  1977.0  ...     True   

       mperhighpt  death  deathhgtm          route1  
3577         8849   True       8800  N Col-NE Ridge  
4838         8750   True       8500  N Col-NE Ridge  
15835        8849   True       8200  N Col-NE Ridge  
20451        8550   True       8550  N Col-NE Ridge  

[4 rows x 14 columns]